test/test.js

```diff
@@ -38,6 +38,45 @@ async function bundle(file, globals, {plugins = []} = {}) {
 }
 
 describe("main", () => {
+  it("no globals", () =>
+    withDir(`
+      - entry.js: |
+          import foo from "foo";
+          console.log(foo);
+    `, async resolve => {
+      await assert.rejects(bundle(resolve("entry.js"), null), {
+        name: "TypeError",
+        message: /Missing/
+      });
+    })
+  );
+
+  it("invalid globals", () =>
+    withDir(`
+      - entry.js: |
+          import foo from "foo";
+          console.log(foo);
+    `, async resolve => {
+      await assert.rejects(bundle(resolve("entry.js"), 1), {
+        name: "TypeError",
+        message: /Unexpected type/
+      });
+    })
+  );
+
+  it("function globals", () =>
+    withDir(`
+      - entry.js: |
+          import foo from "foo";
```

Comment:
Let's use a different identifier e.g. `import bar from "foo"`. It ensures that `id` is identical to the importee.